evaluator.measure/src/main/java/org/opencds/cqf/cql/evaluator/measure/common/MeasureEvaluation.java

```diff
@@ -297,6 +290,33 @@ protected void innerEvaluate(MeasureDef measureDef, List<String> subjectIds, Mea
 
         for (GroupDef groupDef : measureDef.getGroups()) {
             evaluateGroup(measureScoring, groupDef, measureDef.getSdes(), subjectIds);
+            validateEvaluatedMeasureCount(measureScoring, groupDef);
+        }
+    }
+
+    private void validateEvaluatedMeasureCount(MeasureScoring measureScoring, GroupDef groupDef) {
+        switch (measureScoring) {
+            case PROPORTION:
+            case RATIO:
+                //the count of denominator + denominator exclusion + denominator exception must be <= the count of initial population.
+                if ((groupDef.get(INITIALPOPULATION) != null ? groupDef.get(INITIALPOPULATION).getResources().size() : 0) <
```

Comment:
If groupDef.get() returns null, it was likely initialized incorrectly. It's not clear why you need all the null checks here.